storage/database/db_manager.go

```diff
@@ -246,15 +261,19 @@ func checkDBEntryConfigRatio() {
 
 // getDBEntryConfig returns a new DBConfig with original DBConfig and DBEntryType.
 // It adjusts configuration according to the ratio specified in dbConfigRatio and dbDirs.
-func getDBEntryConfig(originalDBC *DBConfig, i DBEntryType) *DBConfig {
+func getDBEntryConfig(originalDBC *DBConfig, i DBEntryType, dbDir string) *DBConfig {
 	newDBC := *originalDBC
 	ratio := dbConfigRatio[i]
 
 	newDBC.LevelDBCacheSize = originalDBC.LevelDBCacheSize * ratio / 100
 	newDBC.OpenFilesLimit = originalDBC.OpenFilesLimit * ratio / 100
 
 	// Update dir to each Database specific directory.
-	newDBC.Dir = filepath.Join(originalDBC.Dir, dbDirs[i])
+	if dbDir == "" {
+		newDBC.Dir = filepath.Join(originalDBC.Dir, dbDirs[i])
+	} else {
+		newDBC.Dir = filepath.Join(originalDBC.Dir, dbDir)
+	}
```

Comment:
We call call `getDBEntryConfig` with `dbDirs[i]` value. 
In that case, we don't need this if branch.